### PR TITLE
Stop propagation when event is synthetic

### LINKED ISSUE #1533: Button type submit triggers click event twice
When pressing on the button and is of type submit and an event listeners is listening it, the callback is triggered twice.

This bug is Blocking. It created 2 form submit on AEM and this can causes problems.

Screenshot 2019-11-25 at 15 51 46

### PR DESCRIPTION
**Pull request by LucaMele**
Fixes #1533 .

# Done is when (DoD):
- [x] I have added UI/Unit tests for my changes
- [ ] I added modified component properties to the typescript typings
- [ ] Design has been reviewed with (here name of reviewer)
- [x] My changes are accordingly to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)
- [x] Old tests and eslint rule are not broken
- [x] My changes are well documented in the README and showcased in the stories 
- [x] Tested with IE

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas (focus on **WHY**)
- [ ] I have made/updated the ChangeLog Section in the README

### CODE REVIEW

File: src/components/10-atoms/button/index.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ import buttonCSS from './index.scss';
 
 const ARROW_RIGHT = 'arrow-right';
 
+// equivalent to event.isTrusted. Unfortunatly IE11 dies not support it
```
Comment (raphaellueckl) [inline src/components/10-atoms/button/index.js]: *does

File: src/components/10-atoms/button/index.js [hunk 2]
```diff
@@ -66,9 +70,14 @@ class AXAButton extends LitElement {
       fakeButton.style.display = 'none';
       this.appendChild(fakeButton);
 
-      // this.onclick refers to the event and has nothing to do with function-valued attribute onClick
-      this.onclick = () => {
-        fakeButton.click();
+      // this click method is triggered due to bubbeling from the button inside
+      // shadowRoot. Do not be confused with onClick
+      this.onclick = e => {
+        // block pr
```
Comment (raphaellueckl) [inline src/components/10-atoms/button/index.js]: block pullrequest?

ahaaaa, propagation :P

**Review by danyball — COMMENTED** [review]
Is it possible to test it? UnitTest for eventIsTrusted should be possible

File: src/components/10-atoms/button/index.js (hunk 1 above)
Comment (LucaMele) [reply src/components/10-atoms/button/index.js]: done

File: src/components/10-atoms/button/index.js (hunk 2 above)
Comment (LucaMele) [reply src/components/10-atoms/button/index.js]: fixed

**Review by LucaMele — COMMENTED** [review]
@danyball i dont think unit test on the method `eventIsTrusted` makes sense but i can test that a click is fired only once. Good point

**Review by 01231 — DISMISSED** [review]

File: src/components/10-atoms/button/index.js [hunk 3]
```diff
@@ -66,9 +70,14 @@ class AXAButton extends LitElement {
       fakeButton.style.display = 'none';
       this.appendChild(fakeButton);
 
-      // this.onclick refers to the event and has nothing to do with function-valued attribute onClick
-      this.onclick = () => {
-        fakeButton.click();
+      // this click method is triggered due to bubbeling from the button inside
+      // shadowRoot. Do not be confused with onClick
+      this.onclick = e => {
+        // block propagation if event is not syntetic
```
Comment (markus-walther) [inline src/components/10-atoms/button/index.js]: spelling: synthetic

File: src/components/10-atoms/button/index.js [hunk 4]
```diff
@@ -10,6 +10,10 @@ import buttonCSS from './index.scss';
 
 const ARROW_RIGHT = 'arrow-right';
 
+// equivalent to event.isTrusted. Unfortunatly IE11 does not support it
```
Comment (markus-walther) [inline src/components/10-atoms/button/index.js]: Spelling and comma: Unfortunately,

File: src/components/10-atoms/button/index.js [hunk 5]
```diff
@@ -10,6 +10,10 @@ import buttonCSS from './index.scss';
...
+const eventIsTrusted = e =>
+  !(!e.screenX && !e.screenY && !e.clientX && !e.clientY);
```
Comment (markus-walther) [inline src/components/10-atoms/button/index.js:16]: Suspicious. Not obvious why this should be equivalent to isTrusted.

I suggest to try a new approach: What if you fire a custom event 'click' on the fakeButton and test for `event instanceOf CustomEvent`? Depends on this still successfully submitting in a form, of course. Worth a try, I think.

**Review by markus-walther — CHANGES_REQUESTED** [review]

File: src/components/10-atoms/button/index.js (hunk 5 above)
Comment (LucaMele) [reply src/components/10-atoms/button/index.js:16]: done

File: src/components/10-atoms/button/index.js (hunk 4 above)
Comment (LucaMele) [reply src/components/10-atoms/button/index.js]: done

File: src/components/10-atoms/button/index.js (hunk 3 above)
Comment (LucaMele) [reply src/components/10-atoms/button/index.js]: done

File: src/components/10-atoms/button/index.js (hunk 5 above)
Comment (markus-walther) [reply src/components/10-atoms/button/index.js:16]: OK to stay in the approach implemented as-is, after our discussion

**Review by markus-walther — DISMISSED** [review]
LGTM

File: src/components/10-atoms/button/ui.test.js [hunk 6]
```diff
@@ -98,3 +98,17 @@ test('should set className', async t => {
 
   await t.expect(await $axaButton.getAttribute('class')).eql('myCssClass');
 });
+
+fixture('Button - Works in a form').page(
+  `${host}/iframe.html?id=atoms-button-demos--feature-button-in-a-form`
+);
+
+test('should submit only once', async t => {
```
Comment (danyball) [inline src/components/10-atoms/button/ui.test.js:106]: nice!

**Review by danyball — DISMISSED** [review]

**Review by danyball — APPROVED** [review]

**Review by markus-walther — APPROVED** [review]
LGTM

**Pull request merged**